### ISSUE TITLE
Multi-Language highlighting

### ISSUE THREAD

**Issue by silverwind** [alecthomas/chroma#783]
**What problem does this feature solve?**

Template languages like Jinja are languages in languages and they would benefit of highlighting twice, once for the templated syntax, and once for the template language itself. Requirement for this to work would be the presence of multiple file extensions like `nginx.conf.j2`.

**What feature do you propose?**

Once as original language, and once as the template language. For example:

- `nginx.conf.j2` - First pass, strip the `.j2` suffix, highlight as nginx
- `nginx.conf.j2` - Second pass, highlight as Jinja, retain nginx highlighting

**Comment by alecthomas** [comment]
Chroma already supports this in the same way as Pygments does, via a [delegating lexer](https://github.com/alecthomas/chroma/blob/master/delegate.go#L12-L20). But I assume what you're proposing is some mechanism by which nested file extensions could be used to do this automatically?

**Comment by silverwind** [comment]
Yes, automatic detection based on filename would be great. I suppose not all languages mix well together, so it should only be enabled for known template languages like jinja, handlebars, etc.
